Multiple sheets source with latest changes

mikel3361 opened the pull request.

mikel3361 committed:
Update cBiff12Container.cls

mikel3361 committed:
Update cBiff12Part.cls

mikel3361 committed:
Update cBiff12Writer.cls

mikel3361 committed:
Create README.md

mikel3361 committed:
Update cBiff12Container.cls

mikel3361 committed:
Update cBiff12Writer.cls

mikel3361 committed:
Update cBiff12Part.cls

mikel3361 committed:
Update cBiff12Writer.cls

wqweto commented on src/cBiff12Writer.cls:
```diff
@@ -130,6 +136,21 @@ Public Function Init( _
+    m_oSheetPart.SheetName = IIf(Not IsMissing(SheetName), SheetName, "Sheet" & Format(m_oContainer.SheetPartCount, "0"))
+    '--- reset previous variables
+    m_lRowCount = 0
+    m_lColCount = Iif(Not IsMissing(lColCount), lColCount, 1)
```
Using `IsMissing` for `Long` optionals is not possible.

Can we just keep current `m_lColCount` when `lColCount` is not supplied with a simple `If` statement?

wqweto commented on src/cBiff12Writer.cls:
```diff
@@ -130,6 +136,21 @@ Public Function Init( _
+Public Function AddSheet(Optional ByVal SheetName As String, Optional ByVal lColCount As Long) As Boolean
+    pvEndSheet m_oSheetPart
+    Set m_oSheetPart = m_oContainer.SheetPart(m_oContainer.SheetPartCount + 1)
+    m_oSheetPart.SheetName = IIf(Not IsMissing(SheetName), SheetName, "Sheet" & Format(m_oContainer.SheetPartCount, "0"))
```
Using `IsMissing` for `String` optional param is not possible (works only w/ `Variant` params).

A `IIf(LenB(SheetName) = 0, ...` can be used for the intended optional string not supplied check.

mikel3361 replied:
I need  my IDE  ...
I check thus tomorrow at work and then we can merge

mikel3361 committed:
Update cBiff12Writer.cls

Rewrite optional parts SheetName, lColCount. lColCount in AddSheet default to 0 and only if lColCount > 0 then different ColCount, otherwise ColCount from init is used.

mikel3361 commented:
Update AddSheet und Init for SheetName with functions from your Info's Can test it tomorow or you make a test. Wenn this is ok you can merge it.

wqweto commented:
LGTM

wqweto commented:
Closes #1
